public/utils/config_to_schema_utils.ts

```diff
@@ -60,15 +47,35 @@ function indexConfigToSchema(indexConfig: IndexConfig): Schema {
   return yup.object(indexSchemaObj);
 }
 
-// TODO: implement this
 function searchConfigToSchema(
   searchConfig: SearchConfig | undefined
 ): ObjectSchema<any> {
   const searchSchemaObj = {} as { [key: string]: Schema };
-
+  if (searchConfig) {
+    searchSchemaObj['request'] = getFieldSchema('json');
+    searchSchemaObj['enrichRequest'] = processorsConfigToSchema(
+      searchConfig.enrichRequest
+    );
+    searchSchemaObj['enrichResponse'] = processorsConfigToSchema(
+      searchConfig.enrichResponse
+    );
+  }
   return yup.object(searchSchemaObj);
 }
 
+function processorsConfigToSchema(processorsConfig: ProcessorsConfig): Schema {
+  const processorsSchemaObj = {} as { [key: string]: Schema };
+  processorsConfig.processors.forEach((processorConfig) => {
+    const processorSchemaObj = {} as { [key: string]: Schema };
+    processorConfig.fields.forEach((field) => {
+      processorSchemaObj[field.id] = getFieldSchema(field.type);
+    });
+    processorsSchemaObj[processorConfig.id] = yup.object(processorSchemaObj);
+  });
+
+  return yup.object(processorsSchemaObj);
```

Comment:
I could be wrong as I'm not very familiar with the Yup library, had to look up the Yup library documentation about `yup.object` to understand what we're trying to do here - define the schema. I see there's another method - `yup.object().shape`, that defines the structure and validation rules for an object. Should we change to use `yup.object().shape` here as `yup.object()` just defines the structure without any constraints?

Reply:
The parameters passed in `object` are where the constraints are set - see [documentation](https://github.com/jquense/yup?tab=readme-ov-file#object). Passing verbosely with `shape` achieves the same thing, but prefer the short-form to keep it simpler.